staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -401,15 +411,26 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 			}
 
 			var expectedProxyAuth string
+			var expectedCredentials string
 			if testCase.proxyAuth != nil {
-				encodedCredentials := base64.StdEncoding.EncodeToString([]byte(testCase.proxyAuth.String()))
-				expectedProxyAuth = "Basic " + encodedCredentials
+				expectedPassword, _ := testCase.proxyAuth.Password()
```

Comment:
don't you have to check if the password is set?

Reply:
I didn't look too much , but it will be nice if we can reuse some of the golang std library methods instead of this manual parsing https://pkg.go.dev/net/http#Request.BasicAuth

Reply:
I was curious about the password because in the code above you check that is not empty to generate the credentials

> > 	proxyAuthUser := proxyURL.User.Username()
> 	proxyAuthPassword, proxyAuthPasswordSet := proxyURL.User.Password()
> 	credentials := proxyAuthUser
> 	if proxyAuthPasswordSet {
> 		credentials += ":" + proxyAuthPassword
> 	}

This is what we should do  https://cs.opensource.google/go/go/+/master:src/net/http/transport.go;l=844-856;drc=c81c0279827545b20f81797bb263ae696d9c235f 

in our `func (s *SpdyRoundTripper) proxyAuth(proxyURL *url.URL) string {`